android-components/components/support/utils/src/main/java/mozilla/components/support/utils/DownloadUtils.kt

```diff
@@ -335,26 +335,30 @@ object DownloadUtils {
     /**
      * Compare the filename extension with the mime type and change it if necessary.
      */
-    private fun changeExtension(filename: String, mimeType: String?): String {
-        var extension: String? = null
-        val dotIndex = filename.lastIndexOf('.')
-
-        if (mimeType != null) {
-            val mimeTypeMap = MimeTypeMap.getSingleton()
-            // Compare the last segment of the extension against the mime type.
-            // If there's a mismatch, discard the entire extension.
-            val typeFromExt = mimeTypeMap.getMimeTypeFromExtension(filename.substringAfterLast('.'))
-            if (typeFromExt?.equals(mimeType, ignoreCase = true) != false) {
-                extension = mimeTypeMap.getExtensionFromMimeType(mimeType)?.let { ".$it" }
-                // Check if the extension needs to be changed
-                if (extension != null && filename.endsWith(extension, ignoreCase = true)) {
-                    return filename
-                }
-            }
-        }
-
-        return if (extension != null) {
-            filename.substring(0, dotIndex) + extension
+    @Suppress("ReturnCount")
+    private fun changeExtension(filename: String, providedMimeType: String?): String {
+        if (providedMimeType == null) return filename
+
+        val file = File(filename)
+        val mimeTypeMap = MimeTypeMap.getSingleton()
+        val extensionFromMimeType =
+            mimeTypeMap.getExtensionFromMimeType(providedMimeType) ?: return filename
```

Comment:
nit: If we moved this line above the first return, we could combine them and potentially get rid of the lint suppression:

```
        val extensionFromMimeType =
            mimeTypeMap.getExtensionFromMimeType(providedMimeType) 
       if (providedMimeType == nul || extensionFromMimeType == null) return filename
```